wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueTypeOpCode.java

```diff
@@ -0,0 +1,82 @@
+package com.dylibso.chicory.wasm.types;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public enum ValueTypeOpCode {
+    UNKNOWN(-1),
+    F64(ID.F64),
+    F32(ID.F32),
+    I64(ID.I64),
+    I32(ID.I32),
+    V128(ID.V128),
+    FuncRef(ID.FuncRef),
+    ExternRef(ID.ExternRef),
+    RefNull(ID.RefNull, List.of(WasmEncoding.VARSINT32)),
+    Ref(ID.Ref, List.of(WasmEncoding.VARSINT32));
+
+    private final int opcode;
+
+    // trick: the enum constructor cannot access its own static fields
+    // but can access another class
+    private static final class ValueTypeOpCodes {
+        private ValueTypeOpCodes() {}
+
+        private static final Map<Integer, ValueTypeOpCode> byOpCode = new HashMap<>();
+        private static final Map<Integer, List<WasmEncoding>> signatures = new HashMap<>();
```

Comment:
I understand why you did it, but, in this case we have exactly 2 special cases:
- `Ref`
- `RefNull`

Instead of keeping around an additional Map I'd prefer to directly handle the different encoding explicitly in the `Parser`.